rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1132,6 +1132,40 @@ public Sha256Hash getBtcBlockchainBlockHashAtDepth(int depth) throws BlockStoreE
         return current.getHeader().getHash();
     }
 
+    public Long getBtcTransactionConfirmationsGetCost(Sha256Hash btcBlockHash) {
+        final long BASIC_COST = 13_600;
+        final long STEP_COST = 70;
+
+        // Dynamic cost based on the depth of the block that contains
+        // the transaction. Find such depth first, then calculate
+        // the cost.
+        Context.propagate(btcContext);
+        try {
+            this.ensureBtcBlockStore();
+            final StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+
+            // Block not found, default to basic cost
+            if (block == null) {
+                return BASIC_COST;
+            }
+
+            final int bestChainHeight = getBtcBlockchainBestChainHeight();
+            final int blockDepth = bestChainHeight - block.getHeight();
```

Comment:
I recommend checking result is not < 0 here.

Reply:
:+1: